info-compiler/src/main/java/com/google/sps/webcrawler/WebCrawler.java

```diff
@@ -246,6 +283,20 @@ public void storeInDatabase(String candidateId, NewsArticle newsArticle) {
     datastore.put(newsArticleEntity);
   }
 
+  /**
+   * For testing purposes.
+   */
+  FullEntity storeInDatabase(String candidateId, NewsArticle newsArticle,
```

Comment:
Same here, rewrite above to reuse the tested logic.

Reply:
I've made the switch from Mockito mocks of Datastore to Datastore emulators. The extra testable method is removed. Now I'm directly testing on the real `storeInDatabase()`.